app/services/green_lanes/find_category_assessments_service.rb

```diff
@@ -2,11 +2,15 @@ module GreenLanes
   class FindCategoryAssessmentsService
     class << self
       def call(goods_nomenclature:, geographical_area_id: nil)
-        goods_nomenclature.applicable_measures.flat_map do |measure|
-          CategoryAssessment.filter(regulation_id: measure.measure_generating_regulation_id,
-                                    measure_type_id: measure.measure_type_id,
-                                    geographical_area: geographical_area_id)
-        end
+        CategoryAssessment.all.map { |category_assessment|
+          matches = goods_nomenclature.applicable_measures.select do |measure|
+            category_assessment.match?(regulation_id: measure.measure_generating_regulation_id,
+                                       measure_type_id: measure.measure_type_id,
+                                       geographical_area: geographical_area_id)
+          end
+
+          matches.any? ? [category_assessment, matches] : nil
+        }.compact
```

Comment:
it looks a bit complex,
I was thinking if we split this in two sections:
1. select the categories
2.  add the category assessment (map)   

it would be simpler